src/network/networkd-route.c

```diff
@@ -1055,6 +1055,7 @@ int route_configure(
 
         _cleanup_(sd_netlink_message_unrefp) sd_netlink_message *req = NULL;
         int r, k = 0;
+        Route *nr;
```

Comment:
it the `else` branch further down is entered, i.e. the one where `route_add_and_setup_time()` is false, then `nr` won't be initialized, and you return rubbish to the caller. either initialize the variable to NULL here up top, or in that `else` branch

Reply:
Hm? `route_add_and_setup_time()` sets `nr` when it returns non-negative, i.e. 0 or 1. So, no rubbish is returned to the caller. Or do I misunderstand your point?

Reply:
Lennart meant the conditional with `ordered_set_isempty(route->multipath_routes)`. In the `else` branch, `nr` is not set.

Reply:
But in the `else` branch we have `assert(!ret);`. So, uninitialized `nr` is no problem. No?